### PR TITLE
fix: require child git repos for workspace import

### PR DESCRIPTION
**Pull request by neversettle17-101**
## Summary
- keep workspace import blocked when validation reports no child Git repositories
- show a no-child-repositories prompt and keep Continue visible but disabled
- add renderer tests for empty workspace folders and non-Git child folders

## Tests
1. Empty folder imported as workspace
Screenshot 2026-09-09 at 12 56 49 AM

2. Root repo git initialised/ non-git & no child git repos
Screenshot 2026-09-09 at 1 34 23 AM

3. Root repo git initialised/ non-git & git initalised but non-remote set child repos
Screenshot 2026-09-09 at 1 29 44 AM

- cd frontend && npm test -- CreateProjectFlow.test.tsx
- cd frontend && npm run typecheck
- git diff --check origin/main...HEAD

## Notes
- Built on current origin/main after #5092 was merged.
- Existing unrelated frontend/package-lock.json changes were left unstaged and are not part of this PR.

### CODE REVIEW

File: frontend/src/renderer/components/CreateProjectFlow.tsx [hunk 1]
```diff
@@ -1878,29 +1783,17 @@ function CreateProjectFolderDialog({
...
-	);
-	const requiredSetupReady = displayRepos.every((repo) => {
-		if (repo.requiredActions.length === 0) return true;
-		const approved = repo.requiredActions.every((action) => workspacePreparation[repo.path]?.approvedActions.includes(action));
-		if (!repo.isRepo && !approved) return true;
-		return approved && (!repo.requiredActions.includes("set_remote") || Boolean(workspacePreparation[repo.path]?.remoteUrl.trim()));
-	});
-	const workspaceSetupReady = !isWorkspace || (workspaceNeedsInitializedRepo ? selectedSetupRepos.length > 0 && selectedSetupReady : requiredSetupReady && selectedSetupReady);
+	const workspaceHasNoChildGitRepos = isWorkspace && validation?.blockingErrors.includes("WORKSPACE_CHILD_REPO_REQUIRED");
+	const workspaceRootIsProject = isWorkspace && validation?.root.isRepo === true && validation.root.hasOrigin === true;
```
Comment (ronishrohan) [inline frontend/src/renderer/components/CreateProjectFlow.tsx:1899]: P1: This now treats a root Git repository without an origin as a workspace, but the daemon still returns `nextStep: choose_import_kind` for **any** workspace root where `root.IsRepo` is true (`backend/internal/service/importer/importer.go:232-236`). That bypasses the backend validation contract and changes the established import rule only in the renderer. Please either update the backend rule in this PR or keep this condition as `validation?.root.isRepo === true`.

**Review by neversettle17-101 — COMMENTED** [review]
## What This PR Does

Keeps workspace imports blocked unless there is at least one direct child Git repository that is already ready, while treating a selected root repository with an origin as a project-import candidate instead of a workspace. The renderer now presents a clearer no-child-repo/remote-required state rather than offering inline workspace Git setup.

## Summary of Changes

- Backend workspace validation only redirects root repositories to project import when the root has an origin, and adds coverage for a root repo without origin that contains a valid child repo.
- CreateProjectFlow removes workspace inline Git setup controls, filters workspace lists to Git repos, disables Continue when any child repo needs setup, and offers project import for folders with no child Git repos.
- Updated Sidebar/CreateProjectFlow/e2e expectations and localized workspace import copy for the new blocked states.

## Findings

No blocking correctness issues found in the reviewed diff. Ready from code review.

## Regression Risk

Moderate UX/behavior risk because this intentionally removes the previous inline workspace child setup path; users with child repos missing remotes or commits must repair them outside this flow before continuing. I did not see an accidental path that still proceeds with incomplete child repos.

## Testing

The PR adds backend validation coverage for root repo plus child repo behavior and updates renderer/sidebar/e2e assertions for empty workspaces, plain child folders, and child repos requiring setup. The PR body reports the focused CreateProjectFlow test, frontend typecheck, and diff check, plus screenshots/manual UI evidence. I did not run tests locally per review-only instructions.

**Review by ronishrohan — APPROVED** [review]
Reviewed the workspace-root validation alignment. The frontend and daemon now enforce the same rule, and the no-origin root case has backend coverage. Approved.
